src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CMDLineProgramModule.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+/**
+ * A class for experimenting with cmd line programs (such as bwa and sga) that's not part of GATK (or has Java bindings yet).
+ */
+public abstract class CMDLineProgramModule {
+
+    public CMDLineProgramModule(){  }
+
+    /**
+     * Builds the initial argument list for the command line (e.g. "bwa mem").
+     * @return a list of strings having the names of requested tools.
+     * @throws IOException
+     */
+    abstract ArrayList<String> initializeCommands() throws IOException;
```

Comment:
in API methods, return `List<String>` rather than a specific implementation of List.